scripts/upload-build-metadata.bash

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+set -x
+set -e
+
+# URL for the associated GitHub Release of $model_name.
+release_url=$1
+# Timestamp for the creation of the $release_url.
+release_time=$2
+# Path to the dir containing the spack.{lock,location.json}.
+json_dir=$3
+# Name of the model (or root-sbd) (eg. access-om2)
+model_name=$4
+
+# the rest of the model components for the given $model_name
+# (eg. for access-om2 there would be mom5, cice5, etc...)
+shift 4
+packages=( "$@" )
+
+spack=$(jq \
+  '{
+    version: .spack.version,
+    commit: .spack.commit
+  }' "$json_dir/spack.lock")
+
+model=$(jq \
+  --arg model "$model_name" \
+  --argjson spack "$spack" \
+  '.concrete_specs | to_entries[] | select(.value.name == $model)
+  | {
+      spack_hash: .key,
+      spec: (.value.name + "@" + .value.version),
+      spack_version: $spack
+  }' "$json_dir/spack.lock"
+)
+
+for pkg in "${packages[@]}"; do
+  install_path=$(jq \
+    --arg pkg "$pkg" \
+    'to_entries[] | select(.key | test($pkg)) | .value' \
+    "$json_dir/spack.location.json"
+  )
+
+  component=$(jq \
+    --arg pkg "$pkg" \
+    --arg release_url "$release_url" \
+    --arg release_time "$release_time" \
+    --arg install_path "$install_path" \
+    '.concrete_specs | to_entries[] | select(.value.name == $pkg)
+    | {
+      spack_hash: .key,
+      spec: (.value.name + "@" + .value.version),
+      install_path: $install_path,
+      created_at: $release_time,
+      release_url: $release_url
+    }' "$json_dir/spack.lock"
+  )
+
+  # construction of the entire package.json
+  jq --null-input \
+    --argjson model "$model" \
+    --argjson component "$component" \
+    '{
+      component_build: $component,
+      model_build: $model
+    }' > package.json
+
+  cat package.json
+
+  python ./tools/release_provenance/save_release.py package.json
+  echo "Uploaded $pkg to build DB"
```

Comment:
I think it is better to do the upload as a command in the CD workflow (maybe && with invoking this script).

In which case maybe this would be renamed to generate_build_metadata.bash or similar?

By doing it this way the script can be tested independently of upload. Speaking of which is there value in adding a test for this into the repo?

Reply:
Makes sense to me - I'll rename it and move the packaging logic out once the schema is ready to go

Reply:
It makes it a lot easier to review if there is some test data to run it against.

Reply:
And also the test data stuff too, will add that as well

Reply:
Added in https://github.com/ACCESS-NRI/build-cd/pull/63/commits/4e89c561b90956cc30baa5539ba97aec68865024